src/main/scala/mesosphere/marathon/api/v2/GroupsResource.scala

```diff
@@ -105,14 +105,18 @@ class GroupsResource @Inject() (
                      body: Array[Byte],
                      @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
     withValid(Json.parse(body).as[GroupUpdate]) { groupUpdate =>
-      val effectivePath = groupUpdate.id.map(_.canonicalPath(id.toRootPath)).getOrElse(id.toRootPath)
+      val path = groupUpdate.id.map(_.canonicalPath(id.toRootPath)).getOrElse(id.toRootPath)
 
-      result(groupManager.rootGroup()).findGroup(_.id == effectivePath) match {
+      result(groupManager.rootGroup()).findGroup(_.id == path) match {
         case Some(currentGroup) =>
-          throw ConflictingChangeException(s"Group $effectivePath is already created. Use PUT to change this group.")
+          throw ConflictingChangeException(s"Group $path is already created. Use PUT to change this group.")
         case None =>
-          val (deployment, path) = updateOrCreate(id.toRootPath, groupUpdate, force)
-          deploymentResult(deployment, Response.created(new URI(path.toString)))
+          result(groupManager.rootGroup()).transitiveApps.find(_.id == path) match {
```

Comment:
The call to groupManager.rootGroup() is expensive and you call it twice here. Please reuse the root group from the first call.